File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/BlockBadge.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.notifiers.BlockBadgeNotifier;

public class BlockBadge<DN extends BlockBadgeNotifier, B extends Box> extends AbstractBlockBadge<B> {
    private int value;

    public BlockBadge(B box) {
        super(box);
    }

    public int value() {
        return this.value;
    }

    public BlockBadge<DN, B> value(int value) {
        this.value = value;
        return this;
    }

    public void update(int value) {
        value(value);
        notifier.refresh(value);
    }
}

File: web-ui/src/displays/components/BlockBadge.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractBlockBadge from "../../../gen/displays/components/AbstractBlockBadge";
import BlockBadgeNotifier from "../../../gen/displays/notifiers/BlockBadgeNotifier";
import BlockBadgeRequester from "../../../gen/displays/requesters/BlockBadgeRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class BlockBadge extends AbstractBlockBadge {

	constructor(props) {
		super(props);
		this.notifier = new BlockBadgeNotifier(this);
		this.requester = new BlockBadgeRequester(this);
	};

	refresh = (value) => {
	};
}

export default withStyles(styles, { withTheme: true })(BlockBadge);
DisplayFactory.register("BlockBadge", withStyles(styles, { withTheme: true })(BlockBadge));